src/ansi.rs

```diff
@@ -0,0 +1,1625 @@
+//! ANSI Terminal Stream Parsing.
+
+use core::str;
+
+#[cfg(feature = "no_std")]
+use arrayvec::{ArrayString, ArrayVec};
+use log::{debug, trace};
+
+use crate::{Parser, Perform};
+
+type Line = usize;
+type Column = usize;
+
+#[derive(Debug, Eq, PartialEq, Copy, Clone, Default)]
+pub struct Rgb {
+    pub r: u8,
+    pub g: u8,
+    pub b: u8,
+}
+
+/// Parse colors in XParseColor format.
+fn xparse_color(color: &[u8]) -> Option<Rgb> {
+    if !color.is_empty() && color[0] == b'#' {
+        parse_legacy_color(&color[1..])
+    } else if color.len() >= 4 && &color[..4] == b"rgb:" {
+        parse_rgb_color(&color[4..])
+    } else {
+        None
+    }
+}
+
+/// Parse colors in `rgb:r(rrr)/g(ggg)/b(bbb)` format.
+fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
+    #[cfg(feature = "no_std")]
+    let colors = str::from_utf8(color).ok()?.split('/').collect::<ArrayVec<[_; 4]>>();
+    #[cfg(not(feature = "no_std"))]
+    let colors = str::from_utf8(color).ok()?.split('/').collect::<Vec<_>>();
```

Comment:
Seems like this function can be easily rewritten to not require any of this Vec/ArrayVec?